src/query/shorthand.ts

```diff
@@ -300,3 +303,200 @@ export function fieldDef(encQ: EncodingQuery,
   }
   return fieldAndParams;
 }
+
+const CHANNEL_INDEX = toMap(CHANNELS);
+const AGGREGATE_OP_INDEX = toMap(AGGREGATE_OPS);
+const SINGLE_TIMEUNIT_INDEX = toMap(SINGLE_TIMEUNITS);
+const MULTI_TIMEUNIT_INDEX = toMap(MULTI_TIMEUNITS);
+
+export function parse(shorthand: string): SpecQuery {
+  // TODO(https://github.com/uwdata/compassql/issues/259):
+  // Do not split directly, but use an upgraded version of `getClosingBraceIndex()`
+  let splitShorthand = shorthand.split('|');
+
+  let specQ: SpecQuery = {mark: splitShorthand[0], encodings: [] as EncodingQuery[]};
+
+  for (let i = 1; i < splitShorthand.length; i++) {
+    let part = splitShorthand[i];
+    const splitPart = splitWithTail(part, ':', 1);
+    const splitPartKey = splitPart[0];
+    const splitPartValue = splitPart[1];
+
+    if (CHANNEL_INDEX[splitPartKey] || splitPartKey === '?') {
+      const encQ = shorthandParser.encoding(splitPartKey, splitPartValue);
+      specQ.encodings.push(encQ);
+      continue;
+    }
+
+    if (splitPartKey === 'calculate') {
+      let transformQ: TransformQuery = specQ.transform || {};
```

Comment:
Remove all declaration of `transformQ`

Just say `specQ.transform = specQ.transform || {}` and always use `specQ.transform`

This way you don't have to assign it back to specQ.transform later, which is redundant.